docs/integrations/spark.md

```diff
@@ -81,6 +184,181 @@ df.write.partitionBy("example-column").parquet(s"s3a://${repo}/${branch}/output-
+spark.sparkContext.hadoopConfiguration.set("fs.lakefs.endpoint", "https://s3.lakefs.example.com")
+```
+  </div>
+  <div markdown="1" id="lakefs-config-tabs-xml">
+
+Ensure you load the lakeFS FileSystem into Spark by running it with `--packages` or `--jars`,
+and then add these into a configuration file, e.g. `$SPARK_HOME/conf/hdfs-site.xml`:
+
+```xml
+<?xml version="1.0"?>
+<configuration>
+    <property>
+        <name>fs.s3a.access.key</name>
+        <value>AKIAIOSFODNN7EXAMPLE</value>
+    </property>
+    <property>
+            <name>fs.s3a.secret.key</name>
+            <value>wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY</value>
+    </property>
+    <property>
+        <name>fs.s3a.endpoint</name>
+        <value>https://s3.eu-central-1.amazonaws.com</value>
+    </property>
+    <property>
+        <name>fs.lakefs.impl</name>
+        <value>io.lakefs.LakeFSFileSystem"</value>
+    </property>
+    <property>
+        <name>fs.lakefs.access.key</name>
+        <value>AKIAlakefs12345EXAMPLE"</value>
+    </property>
+    <property>
+        <name>fs.lakefs.secret.key</name>
+        <value>abc/lakefs/1234567bPxRfiCYEXAMPLEKEY"</value>
+    </property>
+    <property>
+        <name>fs.lakefs.endpoint</name>
+        <value>https://s3.lakefs.example.com"</value>
+    </property>
+</configuration>
+```
+  </div>
+</div>
+
+#### Per-bucket and per-repo configuration
+
+As above, S3 allows for per-bucket configuration.  You can use this if:
+
+1. You need to use S3A directly to access data in an S3 outside of lakeFS, _and_
+1. different credentials are required to access data inside that bucket.
+
+Refer to the Hadoop AWS guide on [Configuring different S3 buckets with Per-Bucket
+Configuration](https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/index.html#Configuring_different_S3_buckets_with_Per-Bucket_Configuration).
+
+There is no need for per-repo configurations in lakeFS when all repositories are on the same
+lakeFS server.  If you need to access repositories that are on *multiple* lakeFS servers,
+configure multiple prefixes.  For instance, you might configure both `fs.lakefs.impl` and
+`fs.lakefs2.impl` to be `io.lakefs.LakeFSFileSystem`, place separate endpoints and credentials
+under `fs.lakefs.*` and `fs.lakefs2.*`, and access the two servers using `lakefs://...` and
+`lakefs2://...` URLs.
+
+### Reading Data
+In order for us to access objects in lakeFS we will need to use the lakeFS path conventions:
+
+```
+lakefs://[REPOSITORY]/[BRANCH]/PATH/TO/OBJECT
+```
+
+Here is an example for reading a parquet file from lakeFS to a Spark DataFrame:
+
+```scala
+val repo = "example-repo"
+val branch = "main"
+val dataPath = s"lakefs://${repo}/${branch}/example-path/example-file.parquet"
+
+val df = spark.read.parquet(dataPath)
+```
+
+You can now use this DataFrame like you would normally do.
+
+### Writing Data
+
+Now simply write your results back to a lakeFS path:
```

Comment:
Suggestion: Write an example of how to write a data frame from code.
There are too many 'now' in the writing data section :)

Reply:
This is pretty much a copy of the earlier "write to S3" section.  Not sure why rewrite it, other than to edit away the "nows".  Since this is unrelated to hadoop-lakefs (same thing appears with "s3a://" above), I'm deferring to another issue / PR.